src/topp/PeptideIndexer.cpp

```diff
@@ -998,15 +991,12 @@ class TOPPPeptideIndexer :
       prot_ids[run_idx].setHits(new_protein_hits);
     }
 
-    /// if requested, store target/decoy status of proteins
-    if (getFlag_("annotate_proteins"))
+    // annotate target/decoy status of proteins:
+    for (vector<ProteinIdentification>::iterator id_it = prot_ids.begin(); id_it != prot_ids.end(); ++id_it)
     {
-      for (vector<ProteinIdentification>::iterator id_it = prot_ids.begin(); id_it != prot_ids.end(); ++id_it)
+      for (vector<ProteinHit>::iterator hit_it = id_it->getHits().begin(); hit_it != id_it->getHits().end(); ++hit_it)
       {
-        for (vector<ProteinHit>::iterator hit_it = id_it->getHits().begin(); hit_it != id_it->getHits().end(); ++hit_it)
-        {
-          hit_it->setMetaValue("target_decoy", (protein_is_decoy[hit_it->getAccession()] ? "decoy" : "target"));
-        }
+        hit_it->setMetaValue("target_decoy", (protein_is_decoy[hit_it->getAccession()] ? "decoy" : "target"));
```

Comment:
target+decoy is also a valid annotation. Do you know if this is handled properly here? I am also missing a few tests for "target+decoy"